src/typing_manager.ts

```diff
@@ -52,6 +52,21 @@ export class TypingManager implements Disposable {
         this.disposables.push(commands.registerCommand("vscode-neovim.send", this.onSendCommand));
         this.disposables.push(commands.registerCommand("vscode-neovim.send-blocking", this.onSendBlockingCommand));
         this.disposables.push(commands.registerCommand("vscode-neovim.escape", this.onEscapeKeyCommand));
+        this.disposables.push(
+            commands.registerCommand("vscode-neovim.toggle", () => {
+                this.main.modeManager.neovimToggle = !this.main.modeManager.neovimToggle;
+                let batch: [string, ...unknown[]][];
+                if (!this.main.modeManager.neovimToggle) {
+                    this.logger.debug(`${LOG_PREFIX}: Disable Vscode-neovim`);
+                    batch = [["mode_change", ["insert"]]];
+                } else {
+                    this.logger.debug(`${LOG_PREFIX}: Enable Vscode-neovim`);
+                    batch = [["mode_change", ["normal"]]];
+                }
+                this.main.modeManager.handleRedrawBatch(batch);
+                this.main.cursorManager.handleRedrawBatch(batch);
```

Comment:
Yeah this just seems like a hack. How come the other managers don't want a redraw batch as well? Really all this does is sets cursor to line and disables `type` event binding, which doesn't actually put nvim in insert mode.

Reply:
in modemanger it calls 
``` 
commands.executeCommand("setContext", "neovim.mode", this.mode);
this.eventEmitter.emit("neovimModeChanged", modeName);
``` 
i can't find the reference to _setContext_.

Could be a hack, i mean it's working and my intent was not tell nvim to enter insertmode i'd rather tell the vscode extension to go into "insertmode" which is basically as statet in the docs act like normal vscode.
It's really my first interaction with extension-programming  and my first time typscript and so on. Help is highly appreciated. I tried to figure out how to send via api _startinsert_ didn't find something useful.

Reply:
`setContext` is just used for vscode keybindings - it's what you use when you do `when: "neovim.mode == insert"`.

Tricking the extension to think it's in insert mode without actually placing nvim in insert mode can cause problems and doesn't serve a purpose (I think it will break insert mode bindings as well). If we tell nvim to go to insert mode, both sides will be in agreement. 

I'm not totally sure but it might be something like `client.command('startinsert')`.

Reply:
If this PR actually decoupled nvim from vscode (killed the subprocess, disabled all bindings, killed all listeners, then re-init the buffers upon enable), then it would be ok to have both sides disagree on the state.

But since all this PR does is make you stuck in insert mode, the model where vscode is simply a "window into nvim" should not be broken.

Reply:
Btw I believe one use of `this.eventEmitter.emit("neovimModeChanged", modeName);` is that it calls `cursorManager.modeChanged` which sets the cursor to be a line instead of block. It also calls `typingManager.modeChanged` which unregisters `type` event.

Reply:
> I'm not totally sure but it might be something like client.command('startinsert').

It works. can you tell me where to find references or the related code for this. I was really stuck and feel stupid now

Reply:
It's tricky. Most of the commands you can find in vim help. `:h startinsert`. It's just a normal ex-command.
Node-client docs are pretty bad, but that is where you find `command` to call a command. https://neovim.io/node-client/classes/NeovimClient.html
Also try to use vscode typescript go to definition for random stuff in the codebase.
For specific ui commands there is `:help ui` and `:help api`. The node client is just a wrapper for `api`.

Reply:
thanks alot!!